src/dodal/common/udc_directory_provider.py

```diff
@@ -0,0 +1,25 @@
+from pathlib import Path
+
+from ophyd_async.core import DirectoryInfo
+
+_directory_info: DirectoryInfo
+
+
+def set_directory(directory: Path):
+    """
+    Set the current base directory of the UDC DirectoryProvider.
+    Files will be stored in the 'panda' subdirectory.
+    """
+    global _directory_info
+    _directory_info = DirectoryInfo(
+        root=directory, resource_dir=Path("panda"), prefix="", suffix=""
+    )
+
+
+def get_udc_directory_provider():
+    """Get the singleton instance of the UDC DirectoryProvider"""
+    return _get_directory
+
+
+def _get_directory() -> DirectoryInfo:
+    return _directory_info
```

Comment:
Is there a reason to have a seperate get_udc_directory_provider rather than setting the get_directory_provider in beamline_utils to the UDC implementation?

Reply:
the one in `beamline_utils` requires `UpdatingDirectoryProvider` however the update() method on it doesn't accept a parameter. That  means that we would then have to register some kind of hook to fetch the directory, or just ignore the method entirely.
It seems more straightforward not to implement this method rather than introduce another layer of callbacks.

Reply:
UpdatingDirectoryProvider.update() isn't a fixed signature right now, I'd be happy for it to become **kwargs: it was just intended as a generic pattern that can be updated in some form